File: C++/Leetcode  997.cpp
#include <stdio.h>
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int findJudge(int n, vector<vector<int>>& trust) {

        if (n == 1)
        {
            return 1;
        }
        unordered_map<int,int> trust_map;
        unordered_map<int,int> ptrust_map;
        for (int i = 0; i < trust.size(); i++)
        {
            trust_map[trust[i][1]]++;
            ptrust_map[trust[i][0]]++;
        }

        auto it = trust_map.begin();

        while (it != trust_map.end())
        {
            if (it -> second == n-1 && !ptrust_map[it->first])
            {
                return it -> first;
            }
            it++;
        }
        return -1;
    }
};

File: Python/Leetcode 997.py
from ast import List


class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        if n == 1:
            return 1

        trusted_people = dict()
        trust_people = set()

        print(type(trusted_people))

        for x in trust:
            # print(x[1])
            # prev_value = trusted_people.get(x[1],0)
            trusted_people[x[1]] = trusted_people.get(x[1],0) + 1
            trust_people.add(x[0])
        
        for person,count in trusted_people.items():
            if count == n - 1:
                if not(person in trust_people):
                    return person
        
        return -1

